contracts/modules/grouping/GroupingModule.sol

```diff
@@ -0,0 +1,220 @@
+
+    /// @notice Returns the canonical protocol-wide LicenseRegistry
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    ILicenseRegistry public immutable LICENSE_REGISTRY;
+
+    // keccak256(abi.encode(uint256(keccak256("story-protocol.GroupingModule")) - 1)) & ~bytes32(uint256(0xff));
+    bytes32 private constant GroupingModuleStorageLocation =
+        0x4f35861babcda7cb8a75afddcc0971d8dc0cbbd9d19afddbe94e0dcd72824100;
+
+    /// Constructor
+    /// @param accessController The address of the AccessController contract
+    /// @param ipAssetRegistry The address of the IpAssetRegistry contract
+    /// @param royaltyModule The address of the RoyaltyModule contract
+    /// @param licenseRegistry The address of the LicenseRegistry contract
+    /// @param groupNFT The address of the GroupNFT contract
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor(
+        address accessController,
+        address ipAssetRegistry,
+        address licenseRegistry,
+        address royaltyModule,
+        address groupNFT
+    ) AccessControlled(accessController, ipAssetRegistry) {
+        if (royaltyModule == address(0)) revert Errors.GroupingModule__ZeroRoyaltyModule();
+        if (licenseRegistry == address(0)) revert Errors.GroupingModule__ZeroLicenseRegistry();
+        if (groupNFT == address(0)) revert Errors.GroupingModule__ZeroGroupNFT();
+        if (ipAssetRegistry == address(0)) revert Errors.GroupingModule__ZeroIpAssetRegistry();
+
+        ROYALTY_MODULE = IRoyaltyModule(royaltyModule);
+        GROUP_IP_ASSET_REGISTRY = IGroupIPAssetRegistry(ipAssetRegistry);
+        LICENSE_REGISTRY = ILicenseRegistry(licenseRegistry);
+
+        if (!groupNFT.supportsInterface(type(IGroupNFT).interfaceId)) {
+            revert Errors.GroupingModule__InvalidGroupNFT(groupNFT);
+        }
+        GROUP_NFT = IGroupNFT(groupNFT);
+        _disableInitializers();
+    }
+
+    /// @notice initializer for this implementation contract
+    /// @param accessManager The address of the protocol admin roles contract
+    function initialize(address accessManager) public initializer {
+        if (accessManager == address(0)) {
+            revert Errors.GroupingModule__ZeroAccessManager();
+        }
+        __ReentrancyGuard_init();
+        __UUPSUpgradeable_init();
+        __ProtocolPausable_init(accessManager);
+    }
+
+    /// @notice Registers a Group IPA.
+    /// @param groupPool The address of the group pool.
+    /// @return groupId The address of the newly registered Group IPA.
+    function registerGroup(address groupPool) external whenNotPaused returns (address groupId) {
+        // mint Group NFT
+        uint256 groupNftId = GROUP_NFT.mintGroupNft(msg.sender, address(this));
+        // register Group NFT
+        groupId = GROUP_IP_ASSET_REGISTRY.registerGroup(address(GROUP_NFT), groupNftId, groupPool);
+        // initialize royalty vault
```

Comment:
// initialize royalty vault
// transfer all royalty tokens to the  group pool

Are there two comments still valid?

Reply:
removed the comments.